migration-tools: avoid wide table (#9580)
pingcap/docs: migration-tools.md

> 

Reviewer: Preview: https://migration-tools.pingcap-docs-website-scaffold-1hq.pages.dev/tidb/dev/migration-tools

!TiDB Migration Tools media/migration-tools.png-->

> The following list introduces the user scenarios, the supported upstreams and downstreams of migration tools.

Reviewer 2: The meaning of this sentence is duplicated with L10. Can we remove it and refine the sentence in L10?

Reviewer: updated in [d732e82](https://github.com/pingcap/docs/pull/9580/commits/d732e823e7f7a9bf82dd6ec9cd317e3d62d35160)

## [TiDB Data Migration (DM)](/dm/dm-overview.md)

| User scenario |<span style="font-weight:normal">Data migration from MySQL-compatible databases to TiDB</span>|
|---|---|

> | **Upstream (or the imported source file)** | MySQL, MariaDB, Aurora, MySQL |

Reviewer 2: ```suggestion
| **Upstream** | MySQL, MariaDB, Aurora, MySQL |
```

Reviewer 2: Can DM import source files directly? If not, we can remove "(or the imported source file)".

> | **Downstream (or the output file)** | TiDB |

Reviewer 2: ```suggestion
| **Downstream** | TiDB |
```

| **Advantages** |<ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul> |

> | **Limitation** | Data import speed is roughly the same as that of TiDB Lightning's TiDB-backend, and a lot lower than that of TiDB Lightning's Local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TiB. |

Reviewer 2: ```suggestion
| **Limitation** | Data import speed is roughly the same as that of TiDB Lightning's [logical import mode](/tidb-lightning/tidb-lightning-logical-import-mode.md), and a lot lower than that of TiDB Lightning's [physical import mode](/tidb-lightning/tidb-lightning-physical-import-mode.md). So it is recommended to use DM to migrate full data with a size of less than 1 TiB. |
```

## [Dumpling](/dumpling-overview.md)

| User scenario | <span style="font-weight:normal">Full data export from MySQL or TiDB</span> |
|---|---|

> | **Upstream (or the imported source file)** | MySQL, TiDB |

Reviewer 2: ```suggestion
| **Upstream** | MySQL, TiDB |
```

Reviewer 2: Dumpling does not need the imported source file

> | **Downstream (or the output file)** | SQL, CSV |

Reviewer 2: ```suggestion
| **Downstream (the output file)** | SQL, CSV |
```

| **Advantages** | <ul><li>Support the table-filter feature that enables you to filter data easier</li><li>Support exporting data to Amazon S3</li></ul> |
| **Limitation** | <ul><li>If you want to restore the exported data to a database other than TiDB, it is recommended to use Dumpling.</li><li>If you want to restore the exported data to another TiDB cluster, it is recommended to use Backup & Restore (BR).</li></ul> |

## [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md)

| User scenario | <span style="font-weight:normal">Full data import into TiDB</span> |
|---|---|

> | **Upstream (or the imported source file)** | <ul><li>Files exported from Dumpling</li><li>Parquet files exported by Amazon Aurora or Apache Hive</li><li>CSV files</li><li>Data read from local disks or Amazon S3</li></ul> |

Reviewer 2: ```suggestion
| **Upstream (the imported source file)** | <ul><li>Files exported from Dumpling</li><li>Parquet files exported by Amazon Aurora or Apache Hive</li><li>CSV files</li><li>Data from local disks or Amazon S3</li></ul> |
```

> | **Downstream (or the output file)** | TiDB |

Reviewer 2: ```suggestion
| **Downstream** | TiDB |
```

| **Advantages** | <ul><li>Support quickly importing a large amount of data and quickly initializing a specific table in a TiDB cluster </li><li>Support checkpoints to store the import progress, so that `tidb-lightning` continues importing from where it lefts off after restarting</li><li>Support data filtering</li></ul> |

> | **Limitation** | <ul><li>If Local-backend is used for data import, during the import process, the TiDB cluster cannot provide services.</li><li> If you do not want the TiDB services to be impacted, perform the data import according to TiDB Lightning TiDB-backend.</li></ul> |

Reviewer 3: ```suggestion
| **Limitation** | <ul><li>If physical import mode is used for data import, during the import process, the TiDB cluster cannot provide services.</li><li> If you do not want the TiDB services to be impacted, perform the data import according to TiDB Lightning logical import mode.</li></ul> |
```

## [Backup & Restore (BR)](/br/backup-and-restore-overview.md)

| User scenario | <span style="font-weight:normal">Migrate a large amount of TiDB custer data by backing up and restoring data</span> |
|---|---|

> | **Upstream (or the imported source file)** | TiDB |

Reviewer 2: ```suggestion
| **Upstream** | TiDB |
```

> | **Downstream (or the output file)** | SST, backup.meta files, backup.lock files |

Reviewer 2: ```suggestion
| **Downstream (the output file)** | SST, backup.meta files, backup.lock files |
```

| **Advantages** |<ul><li>Suitable for migrating data to another TiDB cluster</li><li>Support backing up data to an external storage for disaster recovery</li></ul> |
| **Limitation** |<ul><li>When BR restores data to the upstream cluster of TiCDC or Drainer, the restored data cannot be replicated to the downstream by TiCDC or Drainer.</li><li>BR supports operations only between clusters that have the same `new_collations_enabled_on_first_bootstrap` value.</li></ul> |

## [TiCDC](/ticdc/ticdc-overview.md)

> | User scenario | <span style="font-weight:normal">This tool is implemented by pulling TiKV change logs. It can restore data to a consistent state with any upstream TSO, and support other systems to subscribe to data changes.</span> |

Reviewer 2: ```suggestion
| User scenario | <span style="font-weight:normal">This tool is implemented by pulling TiKV change logs. It can restore cluster data to a consistent state with any upstream TSO, and support other systems to subscribe to data changes.</span> |
```

|---|---|

> | **Upstream (or the imported source file)** | TiDB |

Reviewer 2: ```suggestion
| **Upstream** | TiDB |
```

> | **Downstream (or the output file)** | TiDB, MySQL, Kafka, Confluent |

Reviewer 2: ```suggestion
| **Downstream** | TiDB, MySQL, Kafka, Confluent |
```

| **Advantages** | Provide TiCDC Open Protocol |

> | **Limitation** | TiCDC only replicates tables that have at least one valid index. The following scenarios are not supported:<ul><li>the TiKV cluster that uses RawKV alone.</li><li>the DDL operation `CREATE SEQUENCE` and the `SEQUENCE` function in TiDB.</li></ul> |

Reviewer 2: ```suggestion
| **Limitation** | TiCDC only replicates tables that have at least one valid index. The following scenarios are not supported:<ul><li>The TiKV cluster that uses RawKV alone.</li><li>The DDL operation `CREATE SEQUENCE` and the `SEQUENCE` function in TiDB.</li></ul> |
```

## [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md)

| User scenario | <span style="font-weight:normal">Incremental replication between TiDB clusters, such as using one TiDB cluster as the secondary cluster of another TiDB cluster</span> |
|---|---|

> | **Upstream (or the imported source file)** | TiDB |

Reviewer 2: ```suggestion
| **Upstream** | TiDB |
```

| **Downstream (or the output file)** | TiDB, MySQL, Kafka, incremental backup files |
| **Advantages** | Support real-time backup and restore. Back up TiDB cluster data to be restored for disaster recovery |
| **Limitation** | TiDB Binlog is incompatible with some TiDB versions. It is recommended that you use [TiCDC](/ticdc/ticdc-overview.md). |

## [sync-diff-inspector](/sync-diff-inspector/sync-diff-inspector-overview.md)

| User scenario | <span style="font-weight:normal">Comparing data stored in the databases with the MySQL protocol</span> |
|---|---|

> | **Upstream (or the imported source file)** | TiDB, MySQL |

Reviewer 2: ```suggestion
| **Upstream** | TiDB, MySQL |
```

> | **Downstream (or the output file)** | TiDB, MySQL |

Reviewer 2: ```suggestion
| **Downstream** | TiDB, MySQL |
```
